engine/src/main/java/org/terasology/engine/world/block/internal/BlockManagerImpl.java

```diff
@@ -95,7 +95,8 @@ public void initialise(List<String> registeredBlockFamilies,
                         if (id != null) {
                             block.setId(id);
                         } else {
-                            logger.error("Missing id for block {} in provided family {}", block.getURI(), family.get().getURI());
+                            logger.atError().addArgument(() -> block.getURI()).addArgument(() -> family.get().getURI()).
+                                    log("Missing id for block {} in provided family {}");
```

Comment:
`getURI()` is a simple getter method so impact is neglectable even if we call it in the worst case for every block in every block family. Also, this is an error log, so the likelihood of the method call being done "in vain" is very low.
```suggestion
                            logger.error("Missing id for block {} in provided family {}", block.getURI(), family.get().getURI()); //NOPMD
```